src/test/java/com/networknt/schema/JsonWalkTest.java

```diff
@@ -105,8 +117,14 @@ public void testWalkWithDifferentListeners() throws IOException {
                 + "     }"
                 + "}")));
         // This instance of schema contains one listener removed.
-        CollectorContext collectorContext = result.getCollectorContext();
-        collectorContext.reset();
+        jsonSchema1.getCollectorContext().reset();
+
+        // the original (source) collector context object is expected to be empty now
+        assertNull(jsonSchema1.getCollectorContext().get(SAMPLE_WALK_COLLECTOR_TYPE));
+
+        // while the one from the result still should hold the values
+        assertEquals(result.getCollectorContext().get(SAMPLE_WALK_COLLECTOR_TYPE), collectedNode);
```

Comment:
Yeah, it might be an issue if customers have disabled automatic resetting. Alternatively, one could expose a simpler interface via the `ValidationResut` like `result.getCollectedData()` where `CollectedData` simply has a get method.